packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2253,8 +2262,25 @@ class InputDecoration {
   /// null.
   ///
   /// The semantic label can be replaced by providing a [semanticCounterText].
+  ///
+  /// If set to empty string and [counter] isn't specified, will display nothing
```

Comment:
That would be a backwards incompatible change.

Reply:
Hmm I was trying to keep the behavior consistent with how it is now.  Let me know if this makes sense.  Currently, with `maxLines` set to some number, I see the following behavior:

  * `counterText` null: default text shown (i.e. "0/10")
  * `counterText` 'some string': shows 'some string'
  * `counterText` '': shows no counterText (or I guess there is an empty Text widget)

I intend to try to match this behavior, so you would need to set `counterText` to empty string (and don't set `counter`/`counterBuilder`) to show nothing for the counter.  Is that the right approach to avoid a breaking change?

Reply:
Ah I see that we were testing if `counterText` was null before.  However, we were also copying in some default styles that would override a null `counterText` with the default "0/10" sort of string.  So effectively, the only way to not show `counterText` was to pass an empty string.